libvast/src/meta_index.cpp

```diff
@@ -160,13 +144,22 @@ std::vector<uuid> meta_index::lookup(const expression& expr) const {
             };
             return search(pred);
           } else if (lhs.attr == system::type_atom::value) {
+            // We don't have to look into the synopses for type queries, just
+            // at the layout names.
             result_type result;
-            for (auto& [part_id, part_syn] : partition_synopses_)
-              for (auto& pair : part_syn)
-                if (evaluate(pair.first.name(), x.op, d)) {
+            for (auto& [part_id, part_syn] : synopses_) {
+              for (auto& pair : part_syn) {
+                // TODO: provide an overload for view of evaluate() so that
+                // we can use string_view here. Fortunately type names are
+                // short, so we're probably not hitting the allocator due to
+                // SSO.
+                auto type_name = data{pair.first.layout_name};
```

Comment:
Piggy-backing on existing synopses is a bit risky. This lookup fails if the matching type does not contain any fields for which we can create a synopsis. Consider
```
type foo = record{
  bar: string
}
```

Reply:
I'm not sure what you mean by "piggy-backing on existing synopses". The logic is the same as before: as long as *one* synopsis exists for a given type, we have to consider the partition.

For example:
```
type foo = record{
  bar: string,
  ts: time #timestamp
}
```

The first evaluation fails, the second succeeds. Therefor, the partition should be inspected further.

Reply:
I was not aware that the keys are inserted even when `make_synopsis` failed.